fearless/ApplicationLayer/Services/Transfer/NFT/EthereumNftTransferService.swift

```diff
@@ -123,8 +150,16 @@ final class EthereumNftTransferService: BaseEthereumService, NftTransferService
         let senderAddress = try EthereumAddress(rawAddress: self.senderAddress.hexToBytes())
         let address = try EthereumAddress(rawAddress: transfer.receiver.hexToBytes())
         let contractAddress = try EthereumAddress(rawAddress: smartContract.hexToBytes())
-        let contract = ws.Contract(type: GenericERC721Contract.self, address: contractAddress)
-        let transferCall = contract.transferFrom(from: senderAddress, to: address, tokenId: tokenId)
+
+        var transferCall: SolidityInvocation
+        switch transfer.nft.tokenType ?? .erc721 {
+        case .erc721:
+            let contract = ws.Contract(type: GenericERC721Contract.self, address: contractAddress)
+            transferCall = contract.transferFrom(from: senderAddress, to: address, tokenId: tokenId)
+        case .erc1155:
+            let contract = ws.Contract(type: GenericERC1155Contract.self, address: contractAddress)
+            transferCall = contract.safeTransferFrom(from: senderAddress, to: address, tokenId: tokenId, value: 1, data: [])
```

Comment:
you can hardcode amount but let's do it on presentation layer. service layer should take parameter to be flexible